_posts/2025-03-31-zscore-hybrid-search.md

```diff
@@ -0,0 +1,178 @@
+
+### Conclusions
+
+
+Our benchmark experiments highlight the following advantages and trade-offs of Z-score normalization compared to min-max normalization in hybrid search approaches:
+
+**Search quality (measured using NDCG@10 across four datasets)**:
+
+* Z-score normalization shows a modest improvement in search quality, with an average increase of 2.08% in NDCG@10 scores.
+* This suggests that Z-score normalization may provide slightly better relevance in search results compared to the default normalization technique min-max.
+
+
+**Latency impact**:
+
+* Z-score normalization shows a small increase in latency across different percentiles, as shown in the following table:
+
+|Latency percentile	|Percent difference	|
+|---	|---	|
+|p50	|0.70%	|
+|p90	|0.50%	|
+|p99	|0.50%	|
+
+* The positive percentages indicate that Z-score normalization has slightly higher latency compared to min-max normalization, but the differences are minimal (less than 1% on average).
+
+**Trade-offs**:
+
+* There's a slight trade-off between search quality and latency. Z-score normalization offers a improvement in search relevance (2.08% increase in NDCG@10) at the cost of a marginal increase in latency (0.50% to 0.72% across different percentiles).
+
+**Consistency**:
+
+* The impact of Z-score normalization varies across datasets, with some showing more significant improvements in search quality than others.
+* Latency impact is relatively consistent across datasets, with most showing small increases or no change
+
+
+**Overall assessment**:
+
+* Z-score normalization provides a modest improvement in search quality with a negligible impact on latency.
+* The choice between Z-score and min-max normalization may depend on specific use cases, with Z-score potentially being preferred when even improvements in search relevance are valuable and the slight latency increase is acceptable.
+
+These findings suggest that Z-score normalization could be a viable alternative to min-max normalization in hybrid search approaches, particularly in scenarios where optimizing search relevance is a priority and the system can tolerate minimal latency increases
+
+
+
+## What’s next?
+
+We are also expanding OpenSearch’s hybrid search capabilities beyond z score by planning the following enhancements to our normalization framework:
+
+**Custom normalization functions**: Enables you to define your own normalization logic and allows fine-tuning of search result rankings. For more information, see [this issue](https://github.com/opensearch-project/neural-search/issues/994).
+
+These enhancements will provide more control over search result ranking while ensuring reliable and consistent hybrid search outcomes. Stay tuned for more information!
+
+
+
+### References
+
+1. https://www.codecademy.com/article/normalization
+2. https://towardsdatascience.com/hybrid-search-2-0-the-pursuit-of-better-search-ce44d6f20c08/
+3. https://github.com/opensearch-project/neural-search/issues/1209
```

Comment:
Can you put few words description for each item and make them a link, e.g. "[RFC] Z Score Normalization Technique for Normalization Processor" for the last link